src/main.c

```diff
@@ -38,920 +38,855 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-  }
+    if (disp) {
+        XCloseDisplay(disp);
+        disp = NULL;
+    }
 }
 
-
 // It assumes that the local variable 'main.c:Imlib_Image image' is in context
-static void apply_filter_if_required(void)
+static void applyFilterIfRequired(void)
 {
-  if (opt.script != NULL) {
-      imlib_apply_filter(opt.script);
-  }
+    if (opt.script)
+        imlib_apply_filter(opt.script);
 }
 
-int
-main(int argc,
-     char **argv)
+int main(int argc, char** argv)
 {
-  Imlib_Image image;
-  Imlib_Image thumbnail;
-  Imlib_Load_Error im_err;
-  char *filename_im = NULL, *filename_thumb = NULL;
-
-  char *have_extension = NULL;
-
-  time_t t;
-  struct tm *tm;
-
-  options_parse(argc, argv);
-
-  init_x_and_imlib(opt.display, 0);
-
-  atexit(uninit_x_and_imlib);
-
-  if (!opt.output_file) {
-    opt.output_file = strdup("%Y-%m-%d-%H%M%S_$wx$h_scrot.png");
-    opt.thumb_file = strdup("%Y-%m-%d-%H%M%S_$wx$h_scrot-thumb.png");
-  } else {
-    scrot_have_file_extension(opt.output_file, &have_extension);
-  }
-
-
-  if (opt.focused)
-    image = scrot_grab_focused();
-  else if (opt.select)
-    image = scrot_sel_and_grab_image();
-  else if (opt.autoselect)
-    image = scrot_grab_autoselect();
-  else
-  {
-    scrot_do_delay();
-    if (opt.multidisp) {
-      image = scrot_grab_shot_multi();
-    } else if (opt.stack) {
-      image = scrot_grab_stack_windows();
-    } else {
-      image = scrot_grab_shot();
-    }
-  }
-
-  if (opt.note != NULL)
-    scrot_note_draw(image);
-
-  if (!image) {
-    fprintf(stderr, "no image grabbed: %s", strerror(errno));
-    exit(EXIT_FAILURE);
-  }
-
-  time(&t); /* Get the time directly after the screenshot */
-  tm = localtime(&t);
-
-  imlib_context_set_image(image);
-  imlib_image_attach_data_value("quality", NULL, opt.quality, NULL);
-
-  if (!have_extension) {
-    imlib_image_set_format("png");
-  }
-
-  filename_im = im_printf(opt.output_file, tm, NULL, NULL, image);
-  scrot_check_if_overwrite_file(&filename_im);
-
-  apply_filter_if_required();
-
-  imlib_save_image_with_error_return(filename_im, &im_err);
-  if (im_err) {
-    fprintf(stderr, "Saving to file %s failed: %s\n", filename_im, strerror(errno));
-    exit(EXIT_FAILURE);
-  }
-  if (opt.thumb)
-  {
-    int cwidth, cheight;
-    int twidth, theight;
-
-    cwidth = imlib_image_get_width();
-    cheight = imlib_image_get_height();
-
-    /* Geometry based thumb size */
-    if (opt.thumb_width || opt.thumb_height)
-    {
-      if (!opt.thumb_width)
-      {
-        twidth = cwidth * opt.thumb_height / cheight;
-        theight = opt.thumb_height;
-      }
-      else if (!opt.thumb_height)
-      {
-        twidth = opt.thumb_width;
-        theight = cheight * opt.thumb_width / cwidth;
-      }
-      else
-      {
-        twidth = opt.thumb_width;
-        theight = opt.thumb_height;
-      }
+    Imlib_Image image;
+    Imlib_Image thumbnail;
+    Imlib_Load_Error imErr;
+    char *filenameIM = NULL, *filenameThumb = NULL;
```

Comment:
One on each line:
```C
char* filenameIM = NULL;
char* filenameThumb = NULL;
 ```

Reply:
fixed in 04c1b4969d06103ce9aea93ddf7413c75ab5766a